src/extensions/snapshot/snapshot.js

```diff
@@ -0,0 +1,186 @@
+'use strict'
+
+/**
+ * @module extensions/snapshot/snapshot
+ */
+
+const _ = require('lodash')
+const path = require('path')
+const diff = require('jest-diff')
+const jestDiffConstants = require('jest-diff/build/constants')
+const naturalCompare = require('natural-compare')
+const chalk = require('chalk')
+
+const fileSystem = require('./fs')
+
+const EXPECTED_COLOR = chalk.green
+const RECEIVED_COLOR = chalk.red
+
+exports.scenarioRegex = /^[\s]*Scenario:[\s]*(.*[^\s])[\s]*$/
+
+/**
+ * Extract scenarios from a feature file
+ * @param {string} file - Feature file path
+ * @return {Array<string>} - Scenarios names
+ */
+exports.extractScenarios = file => {
+    if (_.isNil(file)) {
+        throw new TypeError(`Invalid feature file ${file}`)
+    }
+
+    const content = fileSystem.getFileContent(file)
+    const linesContent = _.split(content, '\n')
+
+    let result = []
+    _.forEach(linesContent, (lineContent, idx) => {
+        const line = idx + 1
+        const scenarioInfos = this.scenarioRegex.exec(lineContent)
+        if (scenarioInfos) result.push({ line, name: scenarioInfos[1] })
+    })
+
+    return result
+}
+
+/**
+ * Create snapshots prefix that will be used for each snapshot step of a scenario
+ * For example if the scenario name is 'Scenario 1', then prefix will be 'Scenario 1 1'
+ * If then we have in the same file another scenario named 'Scenario 1', it's prefix will be 'Scenario 1 2' to avoid
+ * naming collisions
+ * @param {Array<string>} scenarios - Scenarios names
+ * @return {Object} - Result will follow the pattern : {scenario_line: {name: scenario_name, line: scenario_line, prefix: scenario_snapshots_prefix} }
```

Comment:
In order to describe the resulting Object it's better to use a typedef, a oneliner like this is hard to read